pkg/serverless/invocationlifecycle/init.go

```diff
@@ -189,3 +189,7 @@ func (lp *LifecycleProcessor) initFromLambdaFunctionURLEvent(event events.Lambda
 	lp.addTag(tagFunctionTriggerEventSourceArn, fmt.Sprintf("arn:aws:lambda:%v:%v:url:%v", region, accountID, functionName))
 	lp.addTags(trigger.GetTagsFromLambdaFunctionURLRequest(event))
 }
+
+func (lp *LifecycleProcessor) initFromStepFunctionEvent(event events.StepFunctionEvent) {
+	lp.requestHandler.event = event.Payload
```

Comment:
In every other method in this file, we set

```go
lp.requestHandler.event = event
```

Why are we setting it to `event.Payload` in this case?

Reply:
That was the consequence of a hack to get around the fact that the trace context was wrapped in a `Payload` field, per the instructions here: https://docs.datadoghq.com/serverless/step_functions/installation/?tab=custom#:~:text=The%20JsonMerge%20intrinsic%20function%20merges%20the%20Step%20Functions%20context%20object%20(%24%24)%20with%20the%20original%20Lambda%E2%80%99s%20input%20payload%20(%24).%20Fields%20of%20the%20original%20payload%20overwrite%20the%20Step%20Functions%20context%20object%20if%20their%20keys%20are%20the%20same.

The actual event is a definitionless JSON object, so deserializing it and storing it is going to be slightly tricky. I'll think about it a little more.

Reply:
Personally, I think it's best to store the event in its original form. This way, all the way through the call stack, we can assume that the `event` is the actual event, and not just a piece of it.

As for json deserialization, you've already made that a bit easier by adding definitions to `pkg/serverless/trigger/events/events.go`. I created that file so that during deserialization, we only need to store the keys we're actually going to use. It really cuts down the number of allocs we make.

Reply:
👍

Reply:
I recommend you save the entire event, not just the payload. In addition to the reasons listed above, it makes your logic easier below.